config/string.c

```diff
@@ -186,6 +186,44 @@ static intptr_t string_native_get(const struct ConfigSet *cs, void *var,
   return (intptr_t) str;
 }
 
+/**
+ * string_string_plus_equals - Concat String to a string - Implements ConfigSetType::string_plus_equals
+ */
+static int string_string_plus_equals(const struct ConfigSet *cs, void *var, const struct ConfigDef *cdef,
+                             const char *value, struct Buffer *err)
+{
+  /* Skip if the value is missing or empty string*/
+  if (!value || (value && (value[0] == '\0')))
+    return CSR_SUCCESS | CSR_SUC_NO_CHANGE;
+
+  int rc = CSR_SUCCESS;
+
+  char *str = NULL;
+  const char **var_str = (const char **) var;
+
+  if (mutt_str_equal(*var_str, ""))
+    str = mutt_str_dup(value);
+  else
+    mutt_str_asprintf(&str, "%s%s", *var_str, value);
```

Comment:
```suggestion
  if (*var_str)
    mutt_str_asprintf(&str, "%s%s", *var_str, value);
  else
    str = mutt_str_dup(value);
```

There are two changes here:
- Replace the test with `(!*var_str)` -- empty strings are stored as `NULL`
- Swap the `if` around so that the `true` case is first